cwms-aaa-client/src/main/java/mil/army/usace/hec/cwms/aaa/client/CacKeyManager.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2022
+ * United States Army Corps of Engineers - Hydrologic Engineering Center (USACE/HEC)
+ * All Rights Reserved.  USACE PROPRIETARY/CONFIDENTIAL.
+ * Source may not be released without written approval from HEC
+ */
+
+package mil.army.usace.hec.cwms.aaa.client;
+
+import java.net.Socket;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.Principal;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateParsingException;
+import java.security.cert.X509Certificate;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import javax.net.ssl.X509KeyManager;
+import org.bouncycastle.asn1.DERUTF8String;
+import org.bouncycastle.asn1.DLSequence;
+import org.bouncycastle.asn1.DLTaggedObject;
+import org.bouncycastle.cert.jcajce.JcaX509ExtensionUtils;
+
+final class CacKeyManager implements X509KeyManager {
+    private static final Logger LOGGER = Logger.getLogger(CacKeyManager.class.getName());
+    private static final Pattern EDIPI_PATTERN = Pattern.compile("\\d{16}@mil");
```

Comment:
I would have thought it was all lower case since we haven't had a problem before, but in the screenshot for this Helpdesk ticket, they have an uppercase:

https://www.dau.edu/faq/Documents/LocateDoDCACInfoEDIPIandPIV.pdf

I added case insensitive check along with some tests.